spring-addons-starter-oidc/src/main/java/com/c4_soft/springaddons/security/oidc/starter/reactive/ReactiveConfigurationSupport.java

```diff
@@ -173,7 +174,11 @@ public void handle(ServerWebExchange exchange, Mono<CsrfToken> csrfToken) {
 
         @Override
         public Mono<String> resolveCsrfTokenValue(ServerWebExchange exchange, CsrfToken csrfToken) {
-            final var hasHeader = exchange.getRequest().getHeaders().get(csrfToken.getHeaderName()).stream().filter(StringUtils::hasText).count() > 0;
+            final var csfrTokenHeader = exchange.getRequest().getHeaders().get(csrfToken.getHeaderName());
+            if (csfrTokenHeader == null) {
```

Comment:
I'm not sure that changing the nature of the exception is the best way to go. Maybe we could just prevent the exception and let `ServerCsrfTokenRequestAttributeHandler` (the parent of this bugged `SpaCsrfTokenRequestHandler`) decide what to do with the missing CSRF token?

Something like that:
```java
final var hasHeader = Optional.ofNullable(exchange.getRequest().getHeaders().get(csrfToken.getHeaderName()))
					.orElse(List.of())
					.stream()
					.filter(StringUtils::hasText)
					.count() > 0;
```

Reply:
I see that you have already changed the code. In fact throwing another exception was not a good choice. I'm happy to have helped, even if a little.

Reply:
I haven't merged anything yet. I tried to add a commit to your PR, but didn't find a way to do it simply. I'd prefer not to create my own branch because your commit would be lost and you'd not be recognized as an author, which would be unfair. Identifying an issue and its origin is the most important work.

Would you kindly copy the code snippet I added above (if you are ok it fixes what you report, of course) and push a new commit with it to your repo to update the PR?

Reply:
I added the new commit. I tested the new behavior and the bug seems to be fixed.